assets/src/components/stopCard.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useId } from "react"
+import { Popup } from "react-leaflet"
+import { DirectionId, Stop } from "../schedule"
+import { RoutePill } from "./routePill"
+import StreetViewButton from "./streetViewButton"
+
+const StopCard = ({
+  stop,
+  direction,
+}: {
+  stop: Stop
+  direction?: DirectionId
+}): JSX.Element => {
+  const connectionsLabelId = "stop-card-connections-label-" + useId()
+
+  const connections = stop.connections
+    ? stop.connections
+        // exclude commuter rail
+        .filter((c) => c.type !== 2)
+        .sort((a, b) => {
+          // non-bubs (i.e. rapid transit) routes go before bus
+          if (a.type === 3 && b.type !== 3) {
+            return 1
+          } else if (a.type !== 3 && b.type === 3) {
+            return -1
+          } else {
+            const aNumeric = Number(a.name)
+            const bNumeric = Number(b.name)
+
+            // SL comes before CT, other than that use localeCompare for non-numeric routes
+            if (isNaN(aNumeric) && isNaN(bNumeric)) {
+              if (a.name.match(/^SL*/) && b.name.match(/^CT*/)) {
+                return -1
+              } else if (a.name.match(/^CT*/) && b.name.match(/^SL*/)) {
+                return 1
+              } else {
+                return a.name.localeCompare(b.name)
+              }
+            } else if (isNaN(aNumeric)) {
+              // purely numeric routes come last
+              return -1
+            } else if (isNaN(bNumeric)) {
+              return 1
+            }
+
+            return aNumeric - bNumeric
+          }
+        })
+    : []
+
+  return (
+    <Popup className="m-stop-card" closeButton={false} offset={[-125, 7]}>
+      <div className="m-stop-card__stop-info">
+        <div className="m-stop-card__stop-name">{stop.name}</div>
+        {direction !== undefined && (
+          <div className="m-stop-card__direction">
+            {direction == 1 ? "Inbound" : "Outbound"}
```

Comment:
The direction ID 1 = inbound, 0 = outbound convention is pretty strong in our GTFS (just remember 1 looks like "i" and 0 looks like "o"). That said it's probably better to actually get it from the data than to hardcode that assumption. I'll look into how easy (or not) it is to do that.

Reply:
So via the `RoutesContext` we already have data for all bus routes, including the direction ID to direction name mapping. However, the difficulty with that is making it so that the `StopCard` component knows what route ID to look up. Since that would also be addressed by some of the refactoring discussed in the other thread, I'll add it to that ticket.